src/Api/Helpers/AddDatabaseExtensionMethod.cs

```diff
@@ -41,9 +41,16 @@ public static IServiceCollection AddDatabase(this IServiceCollection services, I
 
         services.AddScoped<ITenantProvider>(sp =>
         {
-            var accountName = sp.GetService<IHttpContextAccessor>()?.HttpContext?.User
+            var context = sp.GetService<IHttpContextAccessor>()?.HttpContext;
+            var accountName = context?.User
                 .FindFirstValue(CustomClaimTypes.AccountName);
 
+

```

Comment:
Also, if we need to do an environment check I would rather get it from `sp.GetRequiredService<IWebHostEnvironment>()` so we can avoid outer scope capture.